lib/tasks/certs.rake

```diff
@@ -37,4 +37,74 @@ namespace :certs do
       exit 1
     end
   end
+
+  # Per https://fpki.idmanagement.gov/tools/fpkigraph/:
+  # >Most CA certificates will also have an SIA extension with a URI to the CA certificates
+  # >that have been issued by that CA.
+  #
+  # This task takes a key_id as an argument and fetches that cert from the Certificate Store.
+  # If the cert exists, the signing cert is fetched from the Certificate Store.
+  # If the signing cert has a subject information access extension with a 'CA Repository' key,
+  # the bundle is downloaded. This bundle should contain all of the certs signed by the expiring
+  # cert's signing cert.
+  #
+  # There isn't a guaranteed way to associate an expiring cert with its replacment(s), so we try to
+  # match on the subject.
+  desc 'Find replacement cert'
+  task :find_replacement, [:key_id] => [:environment] do |t, args|
+    cert_store = CertificateStore.instance
+    cert_store.load_certs!(dir: Rails.root.join('config/certs'))
+    key_id = args[:key_id]
+
+    cert = CertificateStore.instance[key_id]
+    raise "Cert #{key_id} is not in CertificateStore" if cert.nil?
+    signing_cert = CertificateStore.instance[cert.signing_key_id] ||
+      IssuingCaService.fetch_signing_key_for_cert(cert)
+
+    raise "Signing cert #{cert.signing_key_id} is not in store and could not be downloaded" if signing_cert.nil?
+
+    puts "Signing Cert subject information access (SIA) extensions: #{signing_cert.sia.inspect}"
+
+    repository_certs = IssuingCaService.fetch_ca_repository_certs_for_cert(signing_cert)
+    raise "Signing cert #{cert.signing_key_id} is not in store and could not be downloaded" if signing_cert.nil?
+
+    matching_certs = repository_certs.select do |repo_cert|
+      repo_cert.key_id != cert.key_id &&
```

Comment:
I didn't include it because I wasn't sure, do we want to only suggest certs that are `is_ca_capable?`

https://github.com/18F/identity-pki/blob/ddea9444560b1f56b08684719c0e92c1dd555c04/app/models/certificate.rb#L105-L109

Reply:
Yeah, lets a limitation here. We only want CAs

Reply:
done!